### PR TITLE
Handle erros

### PR DESCRIPTION
**Pull request by maxpeterson**
Includes #48 

Add the errors from the API to the form.

### CODE REVIEW

**Comment by maxpeterson** [comment]
@incuna/js please review (https://github.com/incuna/incuna-surveys/pull/51/commits/55026670f983b1b78139a6e7255926bb0d41f8aa only)

File: web/app/scripts/src/directives/form.es.js [hunk 1]
```diff
@@ -54,10 +54,13 @@ module.directive('surveysForm', [
 
                 scope.submit = function () {
                     if (scope.responseUrl) {
-                        // TODO: Handle errors.
                         API.post(scope.responseUrl, scope.model)
                             .then(scope.onSuccess)
-                            .catch(scope.onFailure);
+                            .catch((response) => {
+                                const errors = response && response.data && response.data;
```
Comment (wytrych) [inline web/app/scripts/src/directives/form.es.js]: `response.data` twice

File: web/app/scripts/src/services/fieldsets-parser.es.js [hunk 2]
```diff
@@ -51,6 +51,17 @@ module.service(moduleProperties.componentName, [
 
             return model;
         };
+
+        this.addFieldErros = function (fields, errors) {
+            fields.forEach((fieldset) => {
+                const key = fieldset.templateOptions.id;
+                const fieldsetErros = errors[key];
+                fieldset.fieldGroup.forEach((field) => {
+                    const fieldError = fieldsetErros && fieldsetErros[field.key];
```
Comment (wytrych) [inline web/app/scripts/src/services/fieldsets-parser.es.js]: typoe `fieldsetErros` -> `fieldsetErrors`

File: web/app/scripts/src/services/fieldsets-parser.es.js [hunk 3]
```diff
@@ -51,6 +51,17 @@ module.service(moduleProperties.componentName, [
 
             return model;
         };
+
+        this.addFieldErros = function (fields, errors) {
```
Comment (wytrych) [inline web/app/scripts/src/services/fieldsets-parser.es.js]: typo `Erros`

File: web/tests/fieldsets-parser-spec.js [hunk 4]
```diff
@@ -99,4 +100,22 @@ describe('FieldsetParserService', function () {
         
     });
 
+    describe('addFieldErros object', function () {
```
Comment (wytrych) [inline web/tests/fieldsets-parser-spec.js]: `Erros` everywhere

**Comment by maxpeterson** [comment]
@wytrych updated.

**Comment by wytrych** [comment]
It looks like you have removed all the compiled code by accident in  	5422771

**Comment by maxpeterson** [comment]
@wytrych updated

**Pull request merged**